drivers/media/i2c/imx477.c

```diff
@@ -1300,6 +1300,11 @@ static int imx477_set_frame_length(struct imx477 *imx477, unsigned int val)
 		val >>= 1;
 	}
 
+	if (imx477->long_exp_shift > IMX477_LONG_EXP_SHIFT_MAX) {
+		imx477->long_exp_shift = IMX477_LONG_EXP_SHIFT_MAX;
+		val = IMX477_FRAME_LENGTH_MAX;
+	}
+
```

Comment:
Can this happen?

Called with ```val = (imx477->mode->height + ctrl->val)```.
```ctrl->val``` has a max range of 
```(IMX477_FRAME_LENGTH_MAX << IMX477_LONG_EXP_SHIFT_MAX) - mode->height```, so shouldn't it only ever be possible to get a max of ```IMX477_FRAME_LENGTH_MAX << IMX477_LONG_EXP_SHIFT_MAX```?

Reply:
Yeah, I was just being ultra defensive.  I suppose i can remove this and let the framework do the verification on limits.